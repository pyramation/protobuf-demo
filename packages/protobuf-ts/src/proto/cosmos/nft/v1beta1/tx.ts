// @generated by protobuf-ts 2.4.0,// @generated from protobuf file "cosmos/nft/v1beta1/tx.proto" (package "cosmos.nft.v1beta1", syntax proto3),// tslint:disable
import { ServiceType } from "@protobuf-ts/runtime-rpc";
import type { BinaryWriteOptions } from "@protobuf-ts/runtime";
import type { IBinaryWriter } from "@protobuf-ts/runtime";
import { WireType } from "@protobuf-ts/runtime";
import type { BinaryReadOptions } from "@protobuf-ts/runtime";
import type { IBinaryReader } from "@protobuf-ts/runtime";
import { UnknownFieldHandler } from "@protobuf-ts/runtime";
import type { PartialMessage } from "@protobuf-ts/runtime";
import { reflectionMergePartial } from "@protobuf-ts/runtime";
import { MESSAGE_TYPE } from "@protobuf-ts/runtime";
import { MessageType } from "@protobuf-ts/runtime";
/**
 * MsgSend represents a message to send a nft from one account to another account.
 *
 * @generated from protobuf message cosmos.nft.v1beta1.MsgSend
 */
export interface MsgSend {
    /**
     * class_id defines the unique identifier of the nft classification, similar to the contract address of ERC721
     *
     * @generated from protobuf field: string class_id = 1;
     */
    classId: string;
    /**
     * id defines the unique identification of nft
     *
     * @generated from protobuf field: string id = 2;
     */
    id: string;
    /**
     * sender is the address of the owner of nft
     *
     * @generated from protobuf field: string sender = 3;
     */
    sender: string;
    /**
     * receiver is the receiver address of nft
     *
     * @generated from protobuf field: string receiver = 4;
     */
    receiver: string;
}
/**
 * MsgSendResponse defines the Msg/Send response type.
 *
 * @generated from protobuf message cosmos.nft.v1beta1.MsgSendResponse
 */
export interface MsgSendResponse {
}
// @generated message type with reflection information, may provide speed optimized methods
class MsgSend$Type extends MessageType<MsgSend> {
    constructor() {
        super("cosmos.nft.v1beta1.MsgSend", [
            { no: 1, name: "class_id", kind: "scalar", T: 9 /*ScalarType.STRING*/ },
            { no: 2, name: "id", kind: "scalar", T: 9 /*ScalarType.STRING*/ },
            { no: 3, name: "sender", kind: "scalar", T: 9 /*ScalarType.STRING*/ },
            { no: 4, name: "receiver", kind: "scalar", T: 9 /*ScalarType.STRING*/ }
        ], { "cosmos.msg.v1.signer": ["sender"] });
    }
    create(value?: PartialMessage<MsgSend>): MsgSend {
        const message = { classId: "", id: "", sender: "", receiver: "" };
        globalThis.Object.defineProperty(message, MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            reflectionMergePartial<MsgSend>(this, message, value);
        return message;
    }
    internalBinaryRead(reader: IBinaryReader, length: number, options: BinaryReadOptions, target?: MsgSend): MsgSend {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* string class_id */ 1:
                    message.classId = reader.string();
                    break;
                case /* string id */ 2:
                    message.id = reader.string();
                    break;
                case /* string sender */ 3:
                    message.sender = reader.string();
                    break;
                case /* string receiver */ 4:
                    message.receiver = reader.string();
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message: MsgSend, writer: IBinaryWriter, options: BinaryWriteOptions): IBinaryWriter {
        /* string class_id = 1; */
        if (message.classId !== "")
            writer.tag(1, WireType.LengthDelimited).string(message.classId);
        /* string id = 2; */
        if (message.id !== "")
            writer.tag(2, WireType.LengthDelimited).string(message.id);
        /* string sender = 3; */
        if (message.sender !== "")
            writer.tag(3, WireType.LengthDelimited).string(message.sender);
        /* string receiver = 4; */
        if (message.receiver !== "")
            writer.tag(4, WireType.LengthDelimited).string(message.receiver);
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message cosmos.nft.v1beta1.MsgSend
 */
export const MsgSend = new MsgSend$Type();
// @generated message type with reflection information, may provide speed optimized methods
class MsgSendResponse$Type extends MessageType<MsgSendResponse> {
    constructor() {
        super("cosmos.nft.v1beta1.MsgSendResponse", []);
    }
    create(value?: PartialMessage<MsgSendResponse>): MsgSendResponse {
        const message = {};
        globalThis.Object.defineProperty(message, MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            reflectionMergePartial<MsgSendResponse>(this, message, value);
        return message;
    }
    internalBinaryRead(reader: IBinaryReader, length: number, options: BinaryReadOptions, target?: MsgSendResponse): MsgSendResponse {
        return target ?? this.create();
    }
    internalBinaryWrite(message: MsgSendResponse, writer: IBinaryWriter, options: BinaryWriteOptions): IBinaryWriter {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message cosmos.nft.v1beta1.MsgSendResponse
 */
export const MsgSendResponse = new MsgSendResponse$Type();
/**
 * @generated ServiceType for protobuf service cosmos.nft.v1beta1.Msg
 */
export const Msg = new ServiceType("cosmos.nft.v1beta1.Msg", [
    { name: "Send", options: {}, I: MsgSend, O: MsgSendResponse }
]);
