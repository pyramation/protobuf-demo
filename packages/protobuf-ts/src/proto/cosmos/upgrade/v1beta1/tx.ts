// @generated by protobuf-ts 2.4.0,// @generated from protobuf file "cosmos/upgrade/v1beta1/tx.proto" (package "cosmos.upgrade.v1beta1", syntax proto3),// tslint:disable
import { ServiceType } from "@protobuf-ts/runtime-rpc";
import type { BinaryWriteOptions } from "@protobuf-ts/runtime";
import type { IBinaryWriter } from "@protobuf-ts/runtime";
import { WireType } from "@protobuf-ts/runtime";
import type { BinaryReadOptions } from "@protobuf-ts/runtime";
import type { IBinaryReader } from "@protobuf-ts/runtime";
import { UnknownFieldHandler } from "@protobuf-ts/runtime";
import type { PartialMessage } from "@protobuf-ts/runtime";
import { reflectionMergePartial } from "@protobuf-ts/runtime";
import { MESSAGE_TYPE } from "@protobuf-ts/runtime";
import { MessageType } from "@protobuf-ts/runtime";
import { Plan } from "./upgrade";
/**
 * MsgSoftwareUpgrade is the Msg/SoftwareUpgrade request type.
 *
 * Since: cosmos-sdk 0.46
 *
 * @generated from protobuf message cosmos.upgrade.v1beta1.MsgSoftwareUpgrade
 */
export interface MsgSoftwareUpgrade {
    /**
     * authority is the address of the governance account.
     *
     * @generated from protobuf field: string authority = 1;
     */
    authority: string;
    /**
     * plan is the upgrade plan.
     *
     * @generated from protobuf field: cosmos.upgrade.v1beta1.Plan plan = 2;
     */
    plan?: Plan;
}
/**
 * MsgSoftwareUpgradeResponse is the Msg/SoftwareUpgrade response type.
 *
 * Since: cosmos-sdk 0.46
 *
 * @generated from protobuf message cosmos.upgrade.v1beta1.MsgSoftwareUpgradeResponse
 */
export interface MsgSoftwareUpgradeResponse {
}
/**
 * MsgCancelUpgrade is the Msg/CancelUpgrade request type.
 *
 * Since: cosmos-sdk 0.46
 *
 * @generated from protobuf message cosmos.upgrade.v1beta1.MsgCancelUpgrade
 */
export interface MsgCancelUpgrade {
    /**
     * authority is the address of the governance account.
     *
     * @generated from protobuf field: string authority = 1;
     */
    authority: string;
}
/**
 * MsgCancelUpgradeResponse is the Msg/CancelUpgrade response type.
 *
 * Since: cosmos-sdk 0.46
 *
 * @generated from protobuf message cosmos.upgrade.v1beta1.MsgCancelUpgradeResponse
 */
export interface MsgCancelUpgradeResponse {
}
// @generated message type with reflection information, may provide speed optimized methods
class MsgSoftwareUpgrade$Type extends MessageType<MsgSoftwareUpgrade> {
    constructor() {
        super("cosmos.upgrade.v1beta1.MsgSoftwareUpgrade", [
            { no: 1, name: "authority", kind: "scalar", T: 9 /*ScalarType.STRING*/, options: { "cosmos_proto.scalar": "cosmos.AddressString" } },
            { no: 2, name: "plan", kind: "message", T: () => Plan, options: { "gogoproto.nullable": false } }
        ], { "cosmos.msg.v1.signer": ["authority"] });
    }
    create(value?: PartialMessage<MsgSoftwareUpgrade>): MsgSoftwareUpgrade {
        const message = { authority: "" };
        globalThis.Object.defineProperty(message, MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            reflectionMergePartial<MsgSoftwareUpgrade>(this, message, value);
        return message;
    }
    internalBinaryRead(reader: IBinaryReader, length: number, options: BinaryReadOptions, target?: MsgSoftwareUpgrade): MsgSoftwareUpgrade {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* string authority */ 1:
                    message.authority = reader.string();
                    break;
                case /* cosmos.upgrade.v1beta1.Plan plan */ 2:
                    message.plan = Plan.internalBinaryRead(reader, reader.uint32(), options, message.plan);
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message: MsgSoftwareUpgrade, writer: IBinaryWriter, options: BinaryWriteOptions): IBinaryWriter {
        /* string authority = 1; */
        if (message.authority !== "")
            writer.tag(1, WireType.LengthDelimited).string(message.authority);
        /* cosmos.upgrade.v1beta1.Plan plan = 2; */
        if (message.plan)
            Plan.internalBinaryWrite(message.plan, writer.tag(2, WireType.LengthDelimited).fork(), options).join();
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message cosmos.upgrade.v1beta1.MsgSoftwareUpgrade
 */
export const MsgSoftwareUpgrade = new MsgSoftwareUpgrade$Type();
// @generated message type with reflection information, may provide speed optimized methods
class MsgSoftwareUpgradeResponse$Type extends MessageType<MsgSoftwareUpgradeResponse> {
    constructor() {
        super("cosmos.upgrade.v1beta1.MsgSoftwareUpgradeResponse", []);
    }
    create(value?: PartialMessage<MsgSoftwareUpgradeResponse>): MsgSoftwareUpgradeResponse {
        const message = {};
        globalThis.Object.defineProperty(message, MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            reflectionMergePartial<MsgSoftwareUpgradeResponse>(this, message, value);
        return message;
    }
    internalBinaryRead(reader: IBinaryReader, length: number, options: BinaryReadOptions, target?: MsgSoftwareUpgradeResponse): MsgSoftwareUpgradeResponse {
        return target ?? this.create();
    }
    internalBinaryWrite(message: MsgSoftwareUpgradeResponse, writer: IBinaryWriter, options: BinaryWriteOptions): IBinaryWriter {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message cosmos.upgrade.v1beta1.MsgSoftwareUpgradeResponse
 */
export const MsgSoftwareUpgradeResponse = new MsgSoftwareUpgradeResponse$Type();
// @generated message type with reflection information, may provide speed optimized methods
class MsgCancelUpgrade$Type extends MessageType<MsgCancelUpgrade> {
    constructor() {
        super("cosmos.upgrade.v1beta1.MsgCancelUpgrade", [
            { no: 1, name: "authority", kind: "scalar", T: 9 /*ScalarType.STRING*/, options: { "cosmos_proto.scalar": "cosmos.AddressString" } }
        ], { "cosmos.msg.v1.signer": ["authority"] });
    }
    create(value?: PartialMessage<MsgCancelUpgrade>): MsgCancelUpgrade {
        const message = { authority: "" };
        globalThis.Object.defineProperty(message, MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            reflectionMergePartial<MsgCancelUpgrade>(this, message, value);
        return message;
    }
    internalBinaryRead(reader: IBinaryReader, length: number, options: BinaryReadOptions, target?: MsgCancelUpgrade): MsgCancelUpgrade {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* string authority */ 1:
                    message.authority = reader.string();
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message: MsgCancelUpgrade, writer: IBinaryWriter, options: BinaryWriteOptions): IBinaryWriter {
        /* string authority = 1; */
        if (message.authority !== "")
            writer.tag(1, WireType.LengthDelimited).string(message.authority);
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message cosmos.upgrade.v1beta1.MsgCancelUpgrade
 */
export const MsgCancelUpgrade = new MsgCancelUpgrade$Type();
// @generated message type with reflection information, may provide speed optimized methods
class MsgCancelUpgradeResponse$Type extends MessageType<MsgCancelUpgradeResponse> {
    constructor() {
        super("cosmos.upgrade.v1beta1.MsgCancelUpgradeResponse", []);
    }
    create(value?: PartialMessage<MsgCancelUpgradeResponse>): MsgCancelUpgradeResponse {
        const message = {};
        globalThis.Object.defineProperty(message, MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            reflectionMergePartial<MsgCancelUpgradeResponse>(this, message, value);
        return message;
    }
    internalBinaryRead(reader: IBinaryReader, length: number, options: BinaryReadOptions, target?: MsgCancelUpgradeResponse): MsgCancelUpgradeResponse {
        return target ?? this.create();
    }
    internalBinaryWrite(message: MsgCancelUpgradeResponse, writer: IBinaryWriter, options: BinaryWriteOptions): IBinaryWriter {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message cosmos.upgrade.v1beta1.MsgCancelUpgradeResponse
 */
export const MsgCancelUpgradeResponse = new MsgCancelUpgradeResponse$Type();
/**
 * @generated ServiceType for protobuf service cosmos.upgrade.v1beta1.Msg
 */
export const Msg = new ServiceType("cosmos.upgrade.v1beta1.Msg", [
    { name: "SoftwareUpgrade", options: {}, I: MsgSoftwareUpgrade, O: MsgSoftwareUpgradeResponse },
    { name: "CancelUpgrade", options: {}, I: MsgCancelUpgrade, O: MsgCancelUpgradeResponse }
]);
