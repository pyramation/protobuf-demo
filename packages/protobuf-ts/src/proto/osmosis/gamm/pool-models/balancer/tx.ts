// @generated by protobuf-ts 2.4.0,// @generated from protobuf file "osmosis/gamm/pool-models/balancer/tx.proto" (package "osmosis.gamm.v1beta1", syntax proto3),// tslint:disable
import { ServiceType } from "@protobuf-ts/runtime-rpc";
import type { BinaryWriteOptions } from "@protobuf-ts/runtime";
import type { IBinaryWriter } from "@protobuf-ts/runtime";
import { WireType } from "@protobuf-ts/runtime";
import type { BinaryReadOptions } from "@protobuf-ts/runtime";
import type { IBinaryReader } from "@protobuf-ts/runtime";
import { UnknownFieldHandler } from "@protobuf-ts/runtime";
import type { PartialMessage } from "@protobuf-ts/runtime";
import { reflectionMergePartial } from "@protobuf-ts/runtime";
import { MESSAGE_TYPE } from "@protobuf-ts/runtime";
import { MessageType } from "@protobuf-ts/runtime";
import { PoolAsset } from "./balancerPool";
import { PoolParams } from "./balancerPool";
/**
 * ===================== MsgCreatePool
 *
 * @generated from protobuf message osmosis.gamm.v1beta1.MsgCreateBalancerPool
 */
export interface MsgCreateBalancerPool {
    /**
     * @generated from protobuf field: string sender = 1;
     */
    sender: string;
    /**
     * @generated from protobuf field: osmosis.gamm.v1beta1.PoolParams poolParams = 2;
     */
    poolParams?: PoolParams;
    /**
     * @generated from protobuf field: repeated osmosis.gamm.v1beta1.PoolAsset poolAssets = 3;
     */
    poolAssets: PoolAsset[];
    /**
     * @generated from protobuf field: string future_pool_governor = 4;
     */
    futurePoolGovernor: string;
}
/**
 * @generated from protobuf message osmosis.gamm.v1beta1.MsgCreateBalancerPoolResponse
 */
export interface MsgCreateBalancerPoolResponse {
    /**
     * @generated from protobuf field: uint64 pool_id = 1;
     */
    poolId: bigint;
}
// @generated message type with reflection information, may provide speed optimized methods
class MsgCreateBalancerPool$Type extends MessageType<MsgCreateBalancerPool> {
    constructor() {
        super("osmosis.gamm.v1beta1.MsgCreateBalancerPool", [
            { no: 1, name: "sender", kind: "scalar", T: 9 /*ScalarType.STRING*/, options: { "gogoproto.moretags": "yaml:\"sender\"" } },
            { no: 2, name: "poolParams", kind: "message", T: () => PoolParams, options: { "gogoproto.moretags": "yaml:\"pool_params\"" } },
            { no: 3, name: "poolAssets", kind: "message", repeat: 1 /*RepeatType.PACKED*/, T: () => PoolAsset, options: { "gogoproto.nullable": false } },
            { no: 4, name: "future_pool_governor", kind: "scalar", T: 9 /*ScalarType.STRING*/, options: { "gogoproto.moretags": "yaml:\"future_pool_governor\"" } }
        ]);
    }
    create(value?: PartialMessage<MsgCreateBalancerPool>): MsgCreateBalancerPool {
        const message = { sender: "", poolAssets: [], futurePoolGovernor: "" };
        globalThis.Object.defineProperty(message, MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            reflectionMergePartial<MsgCreateBalancerPool>(this, message, value);
        return message;
    }
    internalBinaryRead(reader: IBinaryReader, length: number, options: BinaryReadOptions, target?: MsgCreateBalancerPool): MsgCreateBalancerPool {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* string sender */ 1:
                    message.sender = reader.string();
                    break;
                case /* osmosis.gamm.v1beta1.PoolParams poolParams */ 2:
                    message.poolParams = PoolParams.internalBinaryRead(reader, reader.uint32(), options, message.poolParams);
                    break;
                case /* repeated osmosis.gamm.v1beta1.PoolAsset poolAssets */ 3:
                    message.poolAssets.push(PoolAsset.internalBinaryRead(reader, reader.uint32(), options));
                    break;
                case /* string future_pool_governor */ 4:
                    message.futurePoolGovernor = reader.string();
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message: MsgCreateBalancerPool, writer: IBinaryWriter, options: BinaryWriteOptions): IBinaryWriter {
        /* string sender = 1; */
        if (message.sender !== "")
            writer.tag(1, WireType.LengthDelimited).string(message.sender);
        /* osmosis.gamm.v1beta1.PoolParams poolParams = 2; */
        if (message.poolParams)
            PoolParams.internalBinaryWrite(message.poolParams, writer.tag(2, WireType.LengthDelimited).fork(), options).join();
        /* repeated osmosis.gamm.v1beta1.PoolAsset poolAssets = 3; */
        for (let i = 0; i < message.poolAssets.length; i++)
            PoolAsset.internalBinaryWrite(message.poolAssets[i], writer.tag(3, WireType.LengthDelimited).fork(), options).join();
        /* string future_pool_governor = 4; */
        if (message.futurePoolGovernor !== "")
            writer.tag(4, WireType.LengthDelimited).string(message.futurePoolGovernor);
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message osmosis.gamm.v1beta1.MsgCreateBalancerPool
 */
export const MsgCreateBalancerPool = new MsgCreateBalancerPool$Type();
// @generated message type with reflection information, may provide speed optimized methods
class MsgCreateBalancerPoolResponse$Type extends MessageType<MsgCreateBalancerPoolResponse> {
    constructor() {
        super("osmosis.gamm.v1beta1.MsgCreateBalancerPoolResponse", [
            { no: 1, name: "pool_id", kind: "scalar", T: 4 /*ScalarType.UINT64*/, L: 0 /*LongType.BIGINT*/, options: { "gogoproto.customname": "PoolID" } }
        ]);
    }
    create(value?: PartialMessage<MsgCreateBalancerPoolResponse>): MsgCreateBalancerPoolResponse {
        const message = { poolId: 0n };
        globalThis.Object.defineProperty(message, MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            reflectionMergePartial<MsgCreateBalancerPoolResponse>(this, message, value);
        return message;
    }
    internalBinaryRead(reader: IBinaryReader, length: number, options: BinaryReadOptions, target?: MsgCreateBalancerPoolResponse): MsgCreateBalancerPoolResponse {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* uint64 pool_id */ 1:
                    message.poolId = reader.uint64().toBigInt();
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message: MsgCreateBalancerPoolResponse, writer: IBinaryWriter, options: BinaryWriteOptions): IBinaryWriter {
        /* uint64 pool_id = 1; */
        if (message.poolId !== 0n)
            writer.tag(1, WireType.Varint).uint64(message.poolId);
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message osmosis.gamm.v1beta1.MsgCreateBalancerPoolResponse
 */
export const MsgCreateBalancerPoolResponse = new MsgCreateBalancerPoolResponse$Type();
/**
 * @generated ServiceType for protobuf service osmosis.gamm.v1beta1.Msg
 */
export const Msg = new ServiceType("osmosis.gamm.v1beta1.Msg", [
    { name: "CreateBalancerPool", options: {}, I: MsgCreateBalancerPool, O: MsgCreateBalancerPoolResponse }
]);
