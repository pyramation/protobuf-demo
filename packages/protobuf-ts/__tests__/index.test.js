import { coin } from '@cosmjs/amino';
import { MsgJoinPool } from '../src/proto/osmosis/gamm/v1beta1/tx';

it('fromPartial', async () => {
  expect(
    MsgJoinPool.create({
      poolId: BigInt('606'),
      sender: 'osmo1f4vxvvvvvvvvvv3luuddddddddddcccccccccc',
      shareOutAmount: '101010101',
      tokenInMaxs: [
        coin(
          10248,
          'ibc/27394FB092D2ECCD56123C74F36E4C1F926001CEADA9CA97EA622B25F41E5EB2'
        ),
        coin(
          64837969,
          'ibc/B9E0A1A524E98BB407D3CED8720EFEFD186002F90C1B1B7964811DD0CCC12228'
        )
      ]
    })
  ).toMatchSnapshot();
});