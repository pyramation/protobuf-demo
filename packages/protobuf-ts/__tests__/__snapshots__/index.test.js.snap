// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`fromPartial 1`] = `
Object {
  "poolId": 606n,
  "sender": "osmo1f4vxvvvvvvvvvv3luuddddddddddcccccccccc",
  "shareOutAmount": "101010101",
  "tokenInMaxs": Array [
    Object {
      "amount": "10248",
      "denom": "ibc/27394FB092D2ECCD56123C74F36E4C1F926001CEADA9CA97EA622B25F41E5EB2",
    },
    Object {
      "amount": "64837969",
      "denom": "ibc/B9E0A1A524E98BB407D3CED8720EFEFD186002F90C1B1B7964811DD0CCC12228",
    },
  ],
}
`;
